core/src/main/java/io/micronaut/core/execution/CompletableFutureExecutionFlowImpl.java

```diff
@@ -80,6 +81,20 @@ public void onComplete(BiConsumer<? super Object, Throwable> fn) {
         });
     }
 
+    @Nullable
+    @Override
+    public ImperativeExecutionFlow<Object> asComplete() {
+        if (stage.isDone()) {
```

Comment:
There is some kind of concept of the context, but it's not implemented for the completable future as there was no use-case for it. But the idea is to have some `ObjectWithContext` which can hold the context and pass it to the reactive if needed.